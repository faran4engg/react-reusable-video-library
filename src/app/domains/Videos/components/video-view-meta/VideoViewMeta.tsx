import { FC, memo } from 'react';
import { VideoViewMetaProps } from './types';
import { SocialShare } from 'app/domains/Common/components/social-share';

const VideoViewMeta: FC<VideoViewMetaProps> = ({
  videoViewUrl,
  watchVideo,
  title,
  showWatchBtn = true,
}) => {
  const socialShareUrl = window.location.href;

  return (
    <div className="flex items-center justify-between px-3 py-1 pb-2 ">
      <span className="text-xs">
        <img
          loading="lazy"
          src={`https://visitor-badge.glitch.me/badge?page_id=${videoViewUrl}`}
          alt="views"
          width="90%"
          height="100%"
        />
      </span>
      {!showWatchBtn && (
        <SocialShare
          url={socialShareUrl}
          title={`*ðŸŽ¥ ${title}* \n`}
          facebook
          twitter
          telegram
        />
      )}
      {showWatchBtn && (
        <button
          onClick={watchVideo}
          className="px-2 py-1 text-xs font-semibold rounded bg-light-mode-content-bg dark:bg-dark-mode-content-bg"
        >
          â–¶ Watch
        </button>
      )}
    </div>
  );
};

export default memo(VideoViewMeta);
