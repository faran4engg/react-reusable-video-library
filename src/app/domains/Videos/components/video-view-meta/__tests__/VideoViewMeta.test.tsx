import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import VideoViewMeta from '../VideoViewMeta';

/*
	beforeEach(() => {
	  jest.spyOn(global.Math, 'random').mockReturnValue(0.123456789);
	});

	afterEach(() => {
	  jest.spyOn(global.Math, 'random').mockRestore();
	});
*/
const renderer = createRenderer();

describe('<VideoViewMeta />', () => {
  it('should render and match the snapshot with watch button', () => {
    renderer.render(<VideoViewMeta videoViewUrl="/path/id" />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should render and match the snapshot w/o watch button', () => {
    renderer.render(
      <VideoViewMeta videoViewUrl="/path/id" showWatchBtn={false} />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
