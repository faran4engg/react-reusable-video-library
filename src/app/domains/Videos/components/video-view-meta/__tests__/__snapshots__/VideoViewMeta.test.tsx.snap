// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoViewMeta /> should render and match the snapshot w/o watch button 1`] = `
<div
  className="flex items-center justify-between px-3 py-1 pb-2 "
>
  <span
    className="text-xs"
  >
    <img
      alt="views"
      height="100%"
      loading="lazy"
      src="https://visitor-badge.glitch.me/badge?page_id=/path/id"
      width="90%"
    />
  </span>
  <SocialShare
    facebook={true}
    telegram={true}
    title="*ðŸŽ¥ undefined* 
"
    twitter={true}
    url="http://localhost/"
  />
</div>
`;

exports[`<VideoViewMeta /> should render and match the snapshot with watch button 1`] = `
<div
  className="flex items-center justify-between px-3 py-1 pb-2 "
>
  <span
    className="text-xs"
  >
    <img
      alt="views"
      height="100%"
      loading="lazy"
      src="https://visitor-badge.glitch.me/badge?page_id=/path/id"
      width="90%"
    />
  </span>
  <button
    className="px-2 py-1 text-xs font-semibold rounded bg-light-mode-content-bg dark:bg-dark-mode-content-bg"
  >
    â–¶ Watch
  </button>
</div>
`;
